pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar;
+
+import java.util.Arrays;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+/**
+ * Inbuilt Array Transformation Functions
+ * The functions can be used as UDFs in Query when added in the FunctionRegistry.
+ * @ScalarFunction annotation is used with each method for the registration
+ *
+ * Example usage:
+ * <code> SELECT reverseArray(array) FROM baseballStats LIMIT 10 </code>
+ */
+public class ArrayFunctions {
+  private ArrayFunctions() {
+
+  }
+
+  /**
+   * @see ArrayUtils#reverse(Object[])
+   * @param value
+   * @return reversed input array in from end to start
+   */
+  @ScalarFunction
+  public static int[] reverseIntArray(int[] value) {
```

Comment:
for every function that we want to add, we'll have to always write fooLong, fooInt, fooDouble, fooFloat, fooString. In the presto-pinot connector we'll have to do additional mapping. Is there no way around that?

Reply:
@npawar I was going through the string, json, and date scalar functions. I don't see a way to make it generic function for all types. @KKcorps can you please comment.

Reply:
Yes, currently that's not possible. We can resolve this in another PR.